### ISSUE TITLE
Copying files/resources from a crate you reference

### ISSUE THREAD

**Issue by larsbergstrom** [rust-lang/cargo#2955]
It comes up often in Servo that there are files in a crate that we reference that we want to copy to our final output directory. e.g.,
- GPU shader files we need to compile on the target machine for webrender
- The various html files in browser.html

Today, we do that via python code named `find_dep_path_newest` here:
https://github.com/servo/servo/blob/master/python/servo/post_build_commands.py#L36-L45

Basically, we grovel the private crates directory for the ones with the right name. Then, since the hashes are unrelated to the cargo version (so far as we can tell), we just "guess" that the newest one is probably the one you want to copy files from, and use that path.

Is there a better way to do this? e.g., I could imagine setting a "BROWSERHTML_RESOURCES_PATH" env var in the build.rs of the bhtml project and then just picking up that env var in the using project? Or having such variables set automatically by Cargo for use later in the build process?

Note that I'm not sure it makes sense for these projects to "publish" the files as part of their own build.rs, otherwise I'd say that you could have them push the files out to a well-known place in the target directory and then have later build steps pick them up there.

cc @alexcrichton @wycats @metajack

**Comment by alexcrichton** [comment]
Yeah I'm not entirely certain what the best route to go here would be, unfortunately. I suspect a custom env var would work best for now or guessing the layout of the target directory (e.g. putting files in something relative to `OUT_DIR`)

**Comment by epage** [comment]
This has a lot of overlap with #13663 and most of the discussion has happened there, so I'm going to close in favor of that.
